src/m365/spo/commands/file/file-sharinginfo-get.ts

```diff
@@ -0,0 +1,171 @@
+import * as chalk from 'chalk';
+import request from '../../../../request';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import {FileSharingPrincipalType } from './FileSharingPrincipalType';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  id?: string;
+  url?: string;
+}
+
+class SpoFileSharinginfoGetCommand extends SpoCommand {
+
+  public get name(): string {
+    return commands.FILE_SHARINGINFO_GET;
+  }
+
+  public get description(): string {
+    return 'Generates the sharing information report for specified file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = (!(!args.options.id)).toString();
+    telemetryProps.url = (!(!args.options.url)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.log(`Retrieving sharing information report for the file...`);
+    }
+
+    this.getneededFileInformation(args, logger)
+      .then((fileInformation: { fileItemId: number; libraryName: string; }): Promise<string> => {
+        if (this.verbose) {
+          logger.log(`Retrieving Sharing information report for the file with item Id  ${fileInformation.fileItemId}`);
+        }
+
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/lists/getbytitle('${fileInformation.libraryName}')/items(${fileInformation.fileItemId})/GetSharingInformation?$select=permissionsInformation&$Expand=permissionsInformation`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+        return request.post(requestOptions)
+      }).then((res: any): void => {
+        if (args.options.output === 'json') {
+          logger.log(res);
+        }
+        else {
+          let fileSharingInfoCollection: any[] = [];
+
+          res.permissionsInformation.links.map((links: any) => {
```

Comment:
That is a big lesson learned and valuable nugget infact. Would definitely ensure that in future developments.

 Now the code looks really beautify with everything defined in the right way how it supposed to be rather than being generic and confuse future contributors 🤟